QoSFilter.setMaxRequests throws NullPointerException

**Jetty version(s)**
9.4.44
**Java version/vendor** `(use: java -version)`
openjdk version "11.0.10" 2021-01-19 LTS

**OS type/version**
macOS 11.6.2

**Description**
I'm trying to use the QosFilter into an existing Spring application but if I try to tune the filter calling `setMaxRequests` at configuration time, it throws the following NullPointer exception:
```
Caused by: java.lang.NullPointerException: null
	at org.eclipse.jetty.servlets.QoSFilter.setMaxRequests(QoSFilter.java:351)
```
this is the code I'm using to create a Spring Bean with the QosFilter:
```
@Bean
public QoSFilter sfdcJettyQualityOfServiceFilter(MyQoSProperties properties) {
    QoSFilter qoSFilter = new QoSFilter();
    qoSFilter.setSuspendMs(properties.getSuspendMs());
    qoSFilter.setWaitMs(properties.getWaitMs());
    qoSFilter.setMaxRequests(properties.getMaxRequests());

    return qoSFilter;
}
```
I think it blows up because `QosFilter.init()` method has not been called yet at that stage and the private property `_passes` is null [here](https://github.com/eclipse/jetty.project/blob/25306366d4b5d2cdea3eea07861a5d3d145683d6/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/QoSFilter.java#L345).

**How to reproduce?**
- creating a SpringBoot application and trying to load the QosFilter in a `@Configuration` annotated class

--- a/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/QoSFilter.java
+++ b/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/QoSFilter.java
@@ -293,7 +293,9 @@ public class QoSFilter implements Filter
      * for the semaphore to become available before suspending a request.
      *
      * @param value wait time (in milliseconds)
+     * @deprecated do not use
      */
+    @Deprecated
     public void setWaitMs(long value)
     {
         _waitMs = value;
@@ -316,7 +318,9 @@ public class QoSFilter implements Filter
      * a request for while waiting for the semaphore to become available.
      *
      * @param value suspend time (in milliseconds)
+     * @deprecated do not use
      */
+    @Deprecated
     public void setSuspendMs(long value)
     {
         _suspendMs = value;
@@ -339,7 +343,9 @@ public class QoSFilter implements Filter
      * at the same time.
      *
      * @param value the number of requests
+     * @deprecated do not use
      */
+    @Deprecated
     public void setMaxRequests(int value)
     {
         _passes = new Semaphore((value - getMaxRequests() + _passes.availablePermits()), true);

--- a/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/QoSFilter.java
+++ b/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/QoSFilter.java
@@ -293,12 +293,12 @@ public class QoSFilter implements Filter
      * for the semaphore to become available before suspending a request.
      *
      * @param value wait time (in milliseconds)
-     * @deprecated do not use
+     * @deprecated use init-param waitMs instead
      */
     @Deprecated
     public void setWaitMs(long value)
     {
-        _waitMs = value;
+        LOG.warn("Setter ignored: use waitMs init-param for QoSFilter");
     }
 
     /**
@@ -318,12 +318,12 @@ public class QoSFilter implements Filter
      * a request for while waiting for the semaphore to become available.
      *
      * @param value suspend time (in milliseconds)
-     * @deprecated do not use
+     * @deprecated use init-param suspendMs instead
      */
     @Deprecated
     public void setSuspendMs(long value)
     {
-        _suspendMs = value;
+        LOG.warn("Setter ignored: use suspendMs init-param for QoSFilter");
     }
 
     /**
@@ -343,13 +343,12 @@ public class QoSFilter implements Filter
      * at the same time.
      *
      * @param value the number of requests
-     * @deprecated do not use
+     * @deprecated use init-param maxRequests instead
      */
     @Deprecated
     public void setMaxRequests(int value)
     {
-        _passes = new Semaphore((value - getMaxRequests() + _passes.availablePermits()), true);
-        _maxRequests = value;
+        LOG.warn("Setter ignored: use maxRequests init-param for QoSFilter instead");
     }
 
     private class QoSAsyncListener implements AsyncListener

